src/compiler/checker.ts

```diff
@@ -32719,6 +32720,14 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                         effectiveArgs.push(syntheticArg);
                     });
                 }
+                else if (constraint && isTupleType(constraint)) {
+                    forEach(getTypeArguments(constraint), (_, i) => {
+                        const isVariable = !!(constraint.target.elementFlags[i] & ElementFlags.Variable);
+                        const syntheticArg = createSyntheticExpression(arg, !isVariable ? getTypeOfPropertyOfType(spreadType, "" + i as __String)! : spreadType,

```

Comment:
Note that this won't help with variable elements at all - there is no way to express a "deferred" type that would later skip some elements from it so a variable element gets synthesized as the whole `spreadType`. It isn't ideal but it also isn't worse than the current baseline. 

An example code that ideally should work but it doesn't:
```ts
interface HasMethod {
  method(second?: number, ...rest: boolean[]): void;
  method2(first?: string, second?: number, ...rest: boolean[]): void;
}

function fn<HasMethodLike extends HasMethod>(
  instance: HasMethodLike,
  ...args: Parameters<HasMethodLike["method"]>
) {
  instance.method2('', ...args);
}
```

Reply:
The best idea I have for fixing that is to add more information to synthetic expressions (like `.skipElementCount`) and then use that information in `checkSyntheticExpression`. This is how currently `checkSyntheticExpression` looks like:
```ts
function checkSyntheticExpression(node: SyntheticExpression): Type {
    return node.isSpread ? getIndexedAccessType(node.type, numberType) : node.type;
}
```